### ISSUE TITLE
Flutter 1.10 [web] asset path contradictions

### ISSUE THREAD

**Issue by GreenAppers** [flutter/flutter#40707]
It seems that "flutter web build" only includes assets that have been listed in pubspec.yaml. 

However it seems that "flutter run -d chrome" only works with assets in the project_root/web/assets directory.

Listing "web/assets/" in pubspec.yaml will include the files in "flutter build web" (and still works with "flutter run -d chrome") however they are not included in the correct path. 

To clean up this situation, more than just bugfixes are required.  We need direction on:

1) Recommend asset and font locations (Is it project/web/assets or project/assets?) (Is it project/web/assets/fonts or project/web/fonts or project/asset/fonts or project/fonts?)

2) How/if to specify those files in pubspec.yaml. (both icon.png and assets/icon.png can work for the same file, and changes the build produced by "flutter build web")

3) What is the recommended location of AssetsManifest.json and FontsManifest.json and how should the paths be specified in that file? Again there are multiple options possible.

4) How should asset paths be specified within flutter code, e.g. for Image.asset()? Again it's possible use 'assets/icon.png' or 'icon.png'.

I have tested all combinations of the available parameters and found nothing satisfactory.  It's difficult to point at what specifically "the problem" is, except to say that it's broken.

**Comment by jonahwilliams** [comment]
The intention is that flutter assets on the web behave identically on run/build and with other platforms. The fact that these are still reachable is a bug that should be fixed.

1. Wherever you want. Generally under a top level `assets/` directory is common.

2. https://flutter.dev/docs/development/ui/assets-and-images

3. Don't create these, the tool should already be building the,

4. The same path as declared in the pubspec.yaml

**Comment by GreenAppers** [comment]
Thanks!  That information is very helpful.

Now I can report that, using the above recommendations, assets aren't displaying in my web app.

My project has a top level assets and fonts directory (exactly the same as in my working mobile app).

My pubspec.yaml has:
```
flutter:
  uses-material-design: true
  assets:
    - assets/
  fonts:
    - family: MartelSans
      fonts:
        - asset: fonts/MartelSans-ExtraLight.ttf
          weight: 200
```

I have removed the AssetManifest.json and FontManifest.json (and I can verify they are being produced by "flutter build web" and contain the intended assets)

In Flutter code, I'm using Image.asset("assets/logo.png") (As works in mobile app. I also tried "logo.png" and "assets/asssets/logo.png" with no luck).

However when running via "flutter run -d chrome" or serving the build produced by "flutter build web", the asset images are not displayed, and an exception is thrown:

Error while trying to load an asset: Failed to load asset at "assets/logo.png" (404)
══╡ EXCEPTION CAUGHT BY IMAGE RESOURCE SERVICE ╞════════════════════════════════════════════════════
The following assertion was thrown resolving an image codec:
Unable to load asset: logo.png

Expected behavior: the logo.png image is displayed in the web app.

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, v1.10.4-pre.56, on Mac OS X 10.14.6 18G84, locale en-US)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 10.3)
[✓] Chrome - develop for the web
[!] Android Studio (version 3.4)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] Connected device (3 available)

! Doctor found issues in 1 category.
USAJFOUTTSM:cruzweb jfoutts$ flutter doctor -v
[✓] Flutter (Channel master, v1.10.4-pre.56, on Mac OS X 10.14.6 18G84, locale en-US)
    • Flutter version 1.10.4-pre.56 at /Users/jfoutts/flutter
    • Framework revision c4c7e033bd (3 hours ago), 2019-09-18 09:44:41 -0400
    • Engine revision fce8be3a68
    • Dart version 2.6.0 (build 2.6.0-dev.0.0 cb80ea7ba9)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/jfoutts/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 10.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.3, Build version 10G8
    • CocoaPods version 1.6.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version 3.4)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

**Comment by GreenAppers** [comment]
Here is a temporary workaround script for my previous described setup.  Unfortunately the fonts are requested twice.  Possibly related to #36859 

```
#!/bin/bash
rm -rf build build.tgz
flutter build web
mv build/web/assets/assets/*.png build/web/assets
mv build/web/assets/fonts build/web
mkdir build/web/assets/fonts
mv build/web/fonts/MaterialIcons* build/web/assets/fonts
tar cvfz build.tgz build
```

**Comment by yjbanov** [comment]
@jonahwilliams Would your recent fix that serves more things out of the web/ directory fix this as well?

**Comment by jonahwilliams** [comment]
That is unrelated. build_runner will serve arbitrary asset files that are available under lib, while flutter requires them to be specified in assets.

**Comment by matthew-carroll** [comment]
@yjbanov have there been any changes in this area? I just created a Flutter project with web support. I've listed my `assets/` directory under the `assets:` key in the pubspec file. No asset is copied to the `/web` directory, no assets are referenced in the `manifest.json` file, and the asset I'm trying to display is getting a 404.

I've tried killing and rebuilding the app with `flutter web build` followed by `flutter run -d chrome`. I'm still not seeing any change with relation to asset inclusion.

Is there anything else I should be trying for this? Also, do we have any CI tests that verify this particular behavior?
